docs/walkthrough.md

```diff
@@ -0,0 +1,204 @@
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# From Model to Deployment
+
+Given a trained model, how do I take it and deploy it at-scale with an
+optimal configuration using Triton Inference Server? This document is
+here to help answer that.
+
+There are a few steps in this process, and there is no single answer for
+everything, but here is the common flow:
+
+1. Is my model compatible with Triton?
+    - If your model falls under one of Triton's [supported backends](https://github.com/triton-inference-server/backend), then we can simply try to deploy the model as described in the [Quickstart](https://github.com/triton-inference-server/server/blob/main/docs/quickstart.md) guide. For the ONNXRuntime, TensorFlow SavedModel, and TensorRT backends, the model configuration can be completely inferred from the model using Triton's [AutoComplete](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#auto-generated-model-configuration) feature, so no `config.pbtxt` file is required. For other backends, refer to the [Minimal Model Configuration](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#minimal-model-configuration) required to get started.
+    - If your model does not come directly from a supported backend, but is executable through Python; you may be interested in writing a simple python script for running your model through the [Python Backend](https://github.com/triton-inference-server/python_backend).
+    - Otherwise, you can create a [Custom Backend](https://github.com/triton-inference-server/backend/blob/main/examples/README.md) for executing your model.
+
+2. Does my model perform well?
```

Comment:
What value/actions does this section provide? It seems like none, since the next step (Model Analyzer) actually searches for the optimal performance/config.

Same comment for [here](https://github.com/triton-inference-server/server/pull/4706/files#diff-2baaf382502aa8c67478520691f377834f23808a4e116e0b4ee824413e639383R137).

Reply:
Hm, this section is a bit bare, thanks for pointing that out. This PA snippet in particular is to give the reader a visual idea of what PA is immediately without having to click into the other docs/links wondering "what is Perf Analyzer?"

Also, I want to tweak this to more clear that it's a sanity test that you can successfully run inference on your served model (if not, take a step back here and check your config, etc.) as model analyzer will fail

As for the latter section using PA, I think the baseline perf number from a quick PA run help to make Model Analyzer shine a little bit, showing that we can find a config for you that boosts your baseline performance significantly. This is also a gentle guide to show users how to check any arbitrary config quickly - maybe they want to add some `optimization` or `parameters` to the `config.pbtxt` and Model Analyzer can't search/optimize and they want a rough idea of the impact.

Reply:
Thanks for clarifying.

I agree that using PA as a sanity test is a value-add--this should be added to the text. I also agree that mentioning PA here to help market PA is a value-add. I also agree that showing baseline numbers with PA could help illustrate the improvements from MA (although MA does show performance numbers from a "default" run, too), which is a value-add--this could also be added to the text.